src/Components/ArticleDetailPage.jsx

```diff
@@ -1,26 +1,43 @@
 import React, {useState, useEffect} from 'react'
 import {useParams} from 'react-router-dom'
+import{fetchArticleById, fetchCommentsByArticleId} from './api-utils.js'
 import axios from 'axios'
 import './ArticleDetailPage.css'
+import CommentList from './CommentList'
 
 function ArticleDetailPage() {
     const { article_id} = useParams()
     const [article, setArticle] = useState(null)
     const [loading, setLoading] = useState(true)
     const [error, setError] = useState(null)
+    const [comments, setComments] = useState([])
```

Comment:
We should have a separate loading state for any api requests we make where we expect to update the display based on it. This is because if we use the same loading state, it will be set to false when one request finishes regardless of if the other has finished or not - this will affect the 'loading' shown on the screen. You will need a separate loading for the comments and for the article. If you don't require comments in the articles component, it may be easier to manage by moving the comments state into the comments list component :)